portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,35 +26,63 @@
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.sps.data.Comment;
 
-
-/** Servlet that returns some example content. */
+/** Servlet that returns data for my portfolio. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private List comments = new ArrayList<String>();
+  private List comments = new ArrayList<Comment>();
+  private static final int DEFAULT_NUM_COMMENTS = 10;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Load from datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    int commentsMaxNum = DEFAULT_NUM_COMMENTS;
+    try {
+      commentsMaxNum = new Integer(request.getParameter("comments-num"));
+    } catch (NumberFormatException e) {
+      System.err.println("Num of comments wasn't well defined.");
+    }
+    
+    comments.clear();
+    for (Entity entity : results.asIterable()) {
+      if (commentsMaxNum <= 0) {
+          break;
+      }
+      String content = (String) entity.getProperty("content");
+      String user = (String) entity.getProperty("user");
+      Comment comment = new Comment(content, user);
+      comments.add(comment);
+      commentsMaxNum -= 1;
+    }
+
     response.setContentType("application/json");
     response.getWriter().write(new Gson().toJson(comments));
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String newComment = request.getParameter("user-comment");
-    if (newComment.isEmpty()) {
+    String comment = request.getParameter("user-comment");
```

Comment:
I find it confusing that `comment` is later written to `content` member. Maybe consolidate naming?

Reply:
done